File: src/main/java/ru/hh/typofixer/Word.java
package ru.hh.typofixer;

import java.util.Map;

public class Word extends Token {
  private final String value;

  public Word(String value) {
    this.value = value;
  }

  @Override
  public int weight(Map<String, Integer> dict) {
    Integer res = dict.get(value);
    return res == null ? 0 : res;
  }

  @Override
  public String getValue(Map<String, Integer> dict, int lowBound) {
    return value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return value;
  }
}

File: src/main/scala/ru/hh/typofixer/Word.scala
package ru.hh.typofixer

class Word(val string: String) extends Token {
  val weight = Dictionary.prod.weight(string)
  val found = weight > 0

  def weight(dict: Dictionary) = weight

  def value(dict: Dictionary, lowBound: Int) = string

  override def toString = "w!" + string
}
